src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -1380,6 +1380,12 @@ private bool IsQueryMatch(string query, string symbolName)
             DocumentFormattingParams formattingParams,
             RequestContext<TextEdit[]> requestContext)
         {
+            if (this.editorSession.AnalysisService == null)
+            {
+                await requestContext.SendErrorAsync("Script analysis is not enabled in this session");
```

Comment:
Does this wait for the error to be dismissed?  I can't remember if the default for `waitForCompletion` is `true` or `false`.

Also does this appear every formatting request or only once per session?  If the former, do we want to make it once per session so that `formatOnSave` isn't obnoxious or is this uncommon enough that we shouldn't worry about it?

Reply:
As far as I know, this just sends an LSP error message back and the client decides how to handle it. But I don't think it blocks beyond sending the message

Reply:
Right right I'm thinking of the custom show error message request we do, nevermind 🙂

Reply:
This might make VSCode show an error every time though...

Reply:
That should be easy to test.